support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -33,29 +37,106 @@ import (
+	PodUIDLabel            string = "UID"
+	PodNamespaceLabel      string = "Namespace"
+	PodServiceAccountLabel string = "ServiceAccount"
+	PodHostnameLabel       string = "Hostname"
+	PodNodeNameLabel       string = "NodeName"
+	DefaultSpiffeIDPath    string = "ns/{{.Pod.Namespace}}/sa/{{.Pod.ServiceAccount}}"
+)
+
+// PodInfo is created for every processed Pod and it holds pod specific information
+type PodInfo struct {
+	ServiceAccount string
+	Namespace      string
+	Name           string
+	UID            types.UID
+	Hostname       string
+	NodeName       string
+}
+
+// IdentityMaps is used for forming the text from the templates
+type IdentityMaps struct {
+	Context map[string]string
+	Pod     PodInfo
 }
 
 // PodReconciler holds the runtime configuration and state of this controller
 type PodReconciler struct {
 	client.Client
-	c PodReconcilerConfig
+	c             PodReconcilerConfig
+	identityTempl *template.Template
 }
 
 // NewPodReconciler creates a new PodReconciler object
-func NewPodReconciler(config PodReconcilerConfig) *PodReconciler {
-	return &PodReconciler{
-		Client: config.Client,
-		c:      config,
+func NewPodReconciler(config PodReconcilerConfig) (*PodReconciler, error) {
+	if config.IdentityTemplate == "" && config.PodAnnotation == "" && config.PodLabel == "" {
+		config.IdentityTemplate = DefaultSpiffeIDPath
 	}
+	if config.IdentityTemplate == "" {
+		return &PodReconciler{
+			Client: config.Client,
+			c:      config,
+		}, nil
+	}
+
+	tmpl, err := template.New("IdentityTemplate").Parse(config.IdentityTemplate)
+	if err != nil {
+		config.Log.WithError(err).WithField("identity_template", config.IdentityTemplate).Error("error parsing identity template")
+		return &PodReconciler{}, err
+	}
+	// While the Context is persitent and can be tested here, PodInfo is dynamic and changes with each Pod update, so using a dummy entry.
+	templateMaps := IdentityMaps{
+		Context: config.Context,
+		Pod: PodInfo{
+			ServiceAccount: "xx",
+			Namespace:      "xx",
+			Name:           "xx",
+			UID:            types.UID("123"),
+			Hostname:       "xx",
+			NodeName:       "xx",
+		},
+	}
+	var sb strings.Builder
+	if err := tmpl.Execute(&sb, templateMaps); err != nil {
+		config.Log.WithError(err).WithFields(logrus.Fields{
+			"identity_template": config.IdentityTemplate,
+			"context":           config.Context,
+		}).Error("Error executing the identity template")
+		return &PodReconciler{}, err
+	}
+	testSpiffeIDPath := sb.String()
+	testSpiffeID := fmt.Sprintf("spiffe://testdomain/%s", testSpiffeIDPath)
+	if err := idutil.CheckIDStringNormalization(testSpiffeID); err != nil {
+		// The format of the template is incorrect and it is resulting in invalid SPIFFE ID paths.
+		config.Log.WithError(err).WithFields(logrus.Fields{
+			"identity_template": config.IdentityTemplate,
+			"context":           config.Context,
+		}).Error("Error validating the identity template components")
+		return &PodReconciler{}, err
```

Comment:
I think it would aid in debugging if we wrap the error from `idutil` to give it a little more context and make it easier to trace down the error to this code path.

```suggestion
		return &PodReconciler{}, fmt.Errorf("validating identity template: %w", err)
```